gns3server/handlers/upload_handler.py

```diff
@@ -58,6 +58,14 @@ def upload(request, response):
             response.redirect("/upload")
             return
 
+        filename = data['file'].filename
+        file_format = filename.split('.')[-1]
+        valid_formats = ['jpeg', 'tif', 'png', 'gif', 'jpg', 'rif', 'bmp', 'bpg']
+
+        if file_format not in valid_formats:
+            response.redirect('/upload')
+            return
```

Comment:
I think it would be better to return HttpResponse with an error so the user knows he made a mistake while uploading the file. Error message explaining what happened would be informative to the user.